R/sanitize_maf_data.R

```diff
@@ -0,0 +1,60 @@
+#' @title Convert mutation data to a shareable format.
+#'
+#' @description `sanitize_maf_data` returns an oncomatrix of patient/gene data indicating only data needed to produce an oncoplot.
+#'
+#' @details Write an oncomatrix from a MAF File for further plotting. This is meant to be run by individuals who have access to data sets to
+#' "sanitize" a subset of data for subsequent use by them or others who don't have permission to access the raw data.
+#' Example: User J has full permissions for ICGC data and has read permissions on a MAF file. User B needs to make some oncoplots
+#' and/or perform some statistical analysis on the frequency and assortment of mutations in that data set but don't need all the details.
+#' User J can run this function on a maf file and provide the path of the output to user B.
+#'
+#' @param mutation_maf_path Provide either the full path to a MAF file.
+#' @param mutation_maf_data Otherwise provide a data frame of the MAF data.
+#' @param output_oncomatrix Optionally provide the path for your sanitized output file (otherwise it writes to the working directory).
+#' @param genes_keep Specify which genes you want to remain in the output. Make sure there are no duplicated elements in the vector.
+#' @param genes_drop Optionally specify which genes to drop (this doesn't mean all other genes will remain. Maftools decides that part).
+#'
+#' @return The full path to the oncomatrix file (a matrix with Variant_Classification or Multi_Hit indicating coding mutation status per patient).
+#'
+#' @import maftools dplyr
+#' @export
+#'
+#' @examples
+#'
+#' safe_oncomatrix_path = sanitize_maf_data(mutation_maf_data = grande_maf,
```

Comment:
Since the Grande maf is now deprecated in GAMBLR.data, this example should be updated to not use this rsource.